### ISSUE TITLE
[Chapter 10] got 'loss:nan' while training the  model using funstional API

### ISSUE THREAD

**Issue by nirjann** [ageron/handson-ml2#398]
That's the code block

```python
input = keras.layers.Input(shape = X_train_std.shape[1:])
hidden1 = keras.layers.Dense(30, activation = 'relu')(input)
hidden2 = keras.layers.Dense(30, activation = 'relu')(hidden1)
concat = keras.layers.Concatenate()([input, hidden2])
output = keras.layers.Dense(1)(concat)
model = keras.models.Model(inputs = [input], outputs = [output])

model.compile( loss = 'mean_squared_error')
history = model.fit(X_train_std, y_train, epochs = 50)
```

Got the output as:

```
Epoch 1/50
413/413 [==============================] - 1s 2ms/step - loss: nan
Epoch 2/50
413/413 [==============================] - 1s 2ms/step - loss: nan
Epoch 3/50
413/413 [==============================] - 1s 2ms/step - loss: nan
Epoch 4/50
413/413 [==============================] - 1s 2ms/step - loss: nan
Epoch 5/50
413/413 [==============================] - 1s 2ms/step - loss: nan
Epoch 6/50
413/413 [==============================] - 1s 2ms/step - loss: nan
Epoch 7/50
413/413 [==============================] - 1s 2ms/step - loss: nan
```

**Comment by ageron** [comment]
Hi @nirjann ,
Thanks for your feedback. I just tried running [notebook 10 on Colab](https://colab.research.google.com/github/ageron/handson-ml2/blob/master/10_neural_nets_with_keras.ipynb), and I got the following output for the code your mentioned:

```
Epoch 1/20
363/363 [==============================] - 1s 2ms/step - loss: 1.9731 - val_loss: 3.3940
Epoch 2/20
363/363 [==============================] - 0s 1ms/step - loss: 0.7638 - val_loss: 0.9360
Epoch 3/20
363/363 [==============================] - 0s 1ms/step - loss: 0.6045 - val_loss: 0.5649
Epoch 4/20
363/363 [==============================] - 1s 1ms/step - loss: 0.5862 - val_loss: 0.5712
Epoch 5/20
363/363 [==============================] - 0s 1ms/step - loss: 0.5452 - val_loss: 0.5045
Epoch 6/20
363/363 [==============================] - 0s 1ms/step - loss: 0.5243 - val_loss: 0.4831
Epoch 7/20
363/363 [==============================] - 0s 1ms/step - loss: 0.5185 - val_loss: 0.4639
Epoch 8/20
363/363 [==============================] - 1s 1ms/step - loss: 0.4947 - val_loss: 0.4638
Epoch 9/20
363/363 [==============================] - 1s 1ms/step - loss: 0.4782 - val_loss: 0.4421
Epoch 10/20
363/363 [==============================] - 0s 1ms/step - loss: 0.4708 - val_loss: 0.4313
Epoch 11/20
363/363 [==============================] - 0s 1ms/step - loss: 0.4585 - val_loss: 0.4345
Epoch 12/20
363/363 [==============================] - 0s 1ms/step - loss: 0.4481 - val_loss: 0.4168
Epoch 13/20
363/363 [==============================] - 0s 1ms/step - loss: 0.4476 - val_loss: 0.4230
Epoch 14/20
363/363 [==============================] - 0s 1ms/step - loss: 0.4361 - val_loss: 0.4047
Epoch 15/20
363/363 [==============================] - 1s 1ms/step - loss: 0.4392 - val_loss: 0.4078
Epoch 16/20
363/363 [==============================] - 0s 1ms/step - loss: 0.4420 - val_loss: 0.3938
Epoch 17/20
363/363 [==============================] - 0s 1ms/step - loss: 0.4277 - val_loss: 0.3952
Epoch 18/20
363/363 [==============================] - 1s 1ms/step - loss: 0.4216 - val_loss: 0.3860
Epoch 19/20
363/363 [==============================] - 0s 1ms/step - loss: 0.4033 - val_loss: 0.3827
Epoch 20/20
363/363 [==============================] - 0s 1ms/step - loss: 0.3939 - val_loss: 0.4054
162/162 [==============================] - 0s 837us/step - loss: 0.4032
```

There's no `nan` here. Here are some potential solutions:
* Ensure you are running the latest libraries, including TensorFlow 2.4.1. I recently updated `environment.yml` and `requirements.txt`, so you can use them to get a clean environment (see the [installation instructions](https://github.com/ageron/handson-ml2#want-to-install-this-project-on-your-own-machine)).
* Ensure you are running the latest code (run `git pull`). I noticed that your code is slightly different from the one in the notebook (for example you use the default optimizer, which is `"rmsprop"`, but the notebook uses SGD.
* Ensure you run the cells in order, or else you may get unexpected results.
* If you still get `nan`, try reducing the learning rate to a tiny amount.
Hope this helps

**Comment by nirjann** [comment]
I found out I hadn't set the learning rate beforehand. That's why the error occurred. Thank you again. Big fan!
